django/core/management/commands/listurls.py

```diff
@@ -0,0 +1,277 @@
+            "--prefix",
+            "-p",
+            dest="prefixes",
+            help="Only list URLs with these prefixes.",
+            nargs="*",
+        )
+
+        parser.add_argument(
+            "--format",
+            "-f",
+            choices=FORMATS,
+            default="aligned",
+            dest="format",
+            help="Formatting style of the output",
+        )
+
+    def handle(self, *args, **options):
+        prefixes = options["prefixes"]
+        url_patterns = self.get_url_patterns(prefixes=prefixes)
+
+        if not url_patterns:
+            raise CommandError("There are no URL patterns that match given prefixes")
+
+        unsorted = options["unsorted"]
+        no_color = options["no_color"]
+        format = options["format"]
+
+        # Apply sorting
+        if not unsorted:
+            url_patterns.sort()
+
+        # Apply colors
+        self.is_color_enabled = (
+            color.supports_color()
+            and (not no_color)
+            and (format not in COLORLESS_FORMATS)
+        )
+
+        if self.is_color_enabled:
+            url_patterns = self.apply_color(url_patterns=url_patterns)
+
+        # Apply formatting
+        url_patterns = self.apply_format(
+            url_patterns=url_patterns,
+            format=format,
+        )
+
+        return url_patterns
+
+    @classmethod
+    def get_url_patterns(cls, prefixes=None):
+        """
+        Returns a list of URL patterns in the project with given prefixes.
+
+        Each object in the returned list is a tuple[str] with 3 elements:
+        (route, view, name)
+        """
+
+        url_patterns = []
+        urlconf = import_module(settings.ROOT_URLCONF)
+
+        for view_func, regex, namespace_list, name in extract_views_from_urlpatterns(
+            urlconf.urlpatterns
+        ):
+            # Route
+            route = simplify_regex(regex)
+
+            # View
+            view = "{}.{}".format(
+                view_func.__module__,
+                getattr(view_func, "__name__", view_func.__class__.__name__),
+            )
+
+            # Name
+            namespace = ""
+
+            if namespace_list:
+                for part in namespace_list:
+                    namespace += part + ":"
+
+            name = namespace + name if name else None
+            name = name or ""
+
+            # Append to the list
+            url_patterns.append((route, view, name))
+
+        # Filter out when prefixes are given but the pattern's route doesn't match
+        if prefixes:
+            url_patterns = [
+                url_pattern
+                for url_pattern in url_patterns
+                if any(url_pattern[0].startswith(prefix) for prefix in prefixes)
+            ]
+
+        return url_patterns
+
+    def apply_color(self, url_patterns):
+        colored_url_patterns = []
+
+        for url_pattern in url_patterns:
+            # Route
+            route = url_pattern[0]
```

Comment:
I think using a namedtuple (or a different data structure) will give us the readability we want here
Something like
```python
URLPattern = namedtuple("URLPattern", ["route", "view", "name"])
```

Then this section would become:
```python
            route = self.style.ROUTE(url_pattern.route)

            module_path, module_name = url_pattern.view.rsplit(".", 1)
            module_name = self.style.VIEW_NAME(module_name)
            view = f"{module_path}.{module_name}"

            name = url_pattern.name
            if name:
                namespace, name = name.rsplit(":", 1) if ":" in name else ("", name)
                name = self.style.NAME(name)
                name = f"{namespace}:{name}" if namespace else name

            url_pattern = URLPattern(route=route, view=view, name=name)
            url_patterns.append(url_pattern)
```

We can become any index reference and it has another advantage that we can remove `as_dict` as this is inbuilt:
```diff
         return formatted_str.getvalue()
 
-    def as_dict(self, url_pattern):
-        return {
-            "route": url_pattern[0],
-            "view": url_pattern[1],
-            "name": url_pattern[2],
-        }
-
     def format_json(self, url_patterns, pretty=False):
         indent = 4 if pretty else None
 
-        url_pattern_dicts = [self.as_dict(url_pattern) for url_pattern in url_patterns]
+        url_pattern_dicts = [url_pattern._asdict() for url_pattern in url_patterns]
```